### ISSUE TITLE
Support modular database backend

### ISSUE THREAD

**Issue by rkeene** [nanocurrency/nano-node#1093]
Support multiple database backends such as LMDB, RocksDB, and maybe SQLite.

**Comment by CraniumDad** [comment]
Good idea.  As I'm a sysadmin, it would ideal if ordered block data and more random chain index data could be separated in different files and directories.   I'm seeing choppy lmdb IO performance as with #227.

**Comment by zhyatt** [comment]
The support is available for modular database backend at this time, so closing this ticket. We currently have lmdb implemented and will open new tickets for other specific DB implementations in the future.

**Comment by zhyatt** [comment]
Reopening as modular database backend is only partially supported at this time.

**Comment by wezrule** [comment]
RocksDB support added in: https://github.com/nanocurrency/nano-node/pull/2197
